integrations/dd-sdk-android-okhttp/src/main/kotlin/com/datadog/android/okhttp/DatadogInterceptor.kt

```diff
@@ -397,6 +417,48 @@ internal constructor(
 
     // endregion
 
+    // region Builder
+
+    /**
+     * A Builder for the [DatadogInterceptor].
+     */
+    open class Builder(tracedHosts: Map<String, Set<TracingHeaderType>>) :
```

Comment:
IMO having a `map` here is unnecessary complication and it should be just `tracedHosts: List<String>`. Asking to provide tracing header types may be too advanced and increases setup curve. Probably default header types will work for 90%+ installations.

Reply:
it will but then we will have to provide the `setTracedHostWithHeaderTypes` method in the `Builder` and that will create some confusion. It will look like this:
```
val builder = Builder(tracedHosts).setFirstPartyHostsWithHeaderTypes(firstPartyHostsWithHeaderTypes)
```
It will look ugly and users will wonder what is the need for the `tracedHosts` in the Builder.

Reply:
> we will have to provide the setTracedHostWithHeaderTypes method in the Builder

It won't be needed if constructor itself will have another overload. APM setup is already quite complicated, we need to have it simpler with reasonable defaults.

Reply:
so we provide overloads to this `Builder` constructor ? I would rather not and just provide a public api `DEFAULT_HEADER_TYPES = setOf(TracingHeaderType.DATADOG, TracingHeaderType.TRACECONTEXT)`
We are trying to run away from overloads. I am opened to any of this options but I would prefer the one without overloads.
@xgouchet @plousada , what do you think ?

Reply:
> We are trying to run away from overloads
It is just because there were too much of them in the original constructor. Basically we need to put what is mandatory in the constructor and cannot be replaced by some default value.

For now it is only the list of the traced hosts, and the rest of the pieces needed to make interceptor work can have a default value, so they go into the methods of the `Builder` - this separation was missing in the original design of the `DatadogInterceptor`. So having an overload just because of the single **mandatory** parameter looks fine to me.